### PR TITLE
IDEX-3114

### PR DESCRIPTION
**Pull request by vparfonov**
Add providers dpend on OS because for Windows we need to locate ext-server.zip, terminal and projects only in user home directory

### CODE REVIEW

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/HostProjectsFolderProviderUnix.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.machine.ext;
+
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+
+/**
+ * //
```
Comment (skabashnyuk) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/HostProjectsFolderProviderUnix.java]: javadoc

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/TerminalServerBindingProviderWinOS.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * MUST be locate in "user.home" directory in case limitation windows+docker
+ *
+ * @author Vitalii Parfonov
+ */
+public class TerminalServerBindingProviderWinOS implements Provider<String> {
+
+    @Override
+    public String get() {
+        String terminalArchivePath = System.getProperty("user.home") + "\\AppData\\Local\\che\\terminal";
+        return terminalArchivePath + ":/mnt/che/terminal";
```
Comment (skabashnyuk) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/TerminalServerBindingProviderWinOS.java]: can it be made in one line?

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerModule.java [hunk 3]
```diff
@@ -56,5 +59,11 @@ protected void configure() {
                         .build(DockerMachineFactory.class));
 
         Multibinder.newSetBinder(binder(), InstanceProvider.class).addBinding().to(DockerInstanceProvider.class);
+
+        if (SystemInfo.isWindows()) {
+            bind(String.class).annotatedWith(Names.named("host.projects.folder")).toProvider(HostProjectsFolderProviderWinOS.class);
```
Comment (skabashnyuk) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerModule.java]: Use full FQN of class in toProvider method

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerNode.java [hunk 4]
```diff
@@ -33,7 +33,7 @@
     private String host;
 
     @Inject
-    public LocalDockerNode(@Named("vfs.local.fs_root_dir") String projectsFolder) throws IOException {
+    public LocalDockerNode(@Named("host.projects.folder") String projectsFolder) throws IOException {
```
Comment (skabashnyuk) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerNode.java]: We aggree that on machine we have "che.projects.root"
may be we should rename this constant to "host.projects.root" or something like this

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/HostProjectsFolderProviderUnix.java (hunk 1 above)
Comment (vparfonov) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/HostProjectsFolderProviderUnix.java]: ok

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/TerminalServerBindingProviderWinOS.java (hunk 2 above)
Comment (vparfonov) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/TerminalServerBindingProviderWinOS.java]: ok

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerModule.java (hunk 3 above)
Comment (vparfonov) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerModule.java]: ok

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/HostProjectsFolderProviderWinOS.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * MUST be locate in "user.home" directory in case limitation windows+docker
+ *
+ * @author Vitalii Parfonov
+ */
+public class HostProjectsFolderProviderWinOS implements Provider<String> {
+
+
+    @Override
+    public String get() {
+        return System.getProperty("user.home") + "\\che\\projects";
```
Comment (skabashnyuk) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/HostProjectsFolderProviderWinOS.java]: Maybe it's safer to return File or Path?

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/TerminalServerBindingProviderWinOS.java (hunk 2 above)
Comment (skabashnyuk) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/TerminalServerBindingProviderWinOS.java]: Maybe it's safer to return File or Path?

Comment (garagatyi) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/TerminalServerBindingProviderWinOS.java]: We need string, so if it returns File we will need to convert it to string

**Comment by garagatyi** [comment]
LGTM

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerNode.java (hunk 4 above)
Comment (vparfonov) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerNode.java]: yes this is better name

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/TerminalServerBindingProviderWinOS.java (hunk 2 above)
Comment (skabashnyuk) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/TerminalServerBindingProviderWinOS.java]: ок

**Pull request merged**